mfr/extensions/tabular/libs/h5py_scipy_tools.py

```diff
@@ -0,0 +1,75 @@
+from collections import OrderedDict
+
+import h5py
+import scipy.io
+
+from mfr.extensions.tabular import exceptions
+from mfr.extensions.tabular.utilities import data_population, header_population
+
+def build_sheets(name, list_data, sheets):
+    # add default header to fix rendering of some data formats
+    header = [i for i in range(1, 1 + len(list_data[0]))]
+    values = data_population(list_data, headers=header)
+    header = header_population(header)
+    sheets[str(name)] = (header, values)
+    return sheets
+
+def mat_v73(fp):
+    """Read and convert a mat v7.3+ file to JSON format using h5py
+    :param fp: File pointer object
+    :return: tuple of table headers and data
+    """
+
+    workbook = h5py.File(fp.name, 'r')
+
+    if not workbook:
+        raise(exceptions.EmptyTableError("Cannot render an empty mat file"))
+
+    sheets = OrderedDict()
+    variables = workbook.items()
+
+    for var in variables:
+        name = var[0]
+        data = var[1]
+        if type(data) is h5py.Dataset:
+            # h5py Uses row-major ordering. this fixes it so it displays like it does in matlab
+            # basically just flip it on its axis
+            list_data = list(zip(*data.value.tolist()))
+            sheets = build_sheets(name, list_data, sheets)
+
+    return sheets
+
+
+def mat_v7(fp):
+    """Read and convert a mat v7.0 and below file to JSON format using scipy.io.matload
+    :param fp: File pointer object
+    :return: tuple of table headers and data
+    """
+    try:
+        workbook = scipy.io.loadmat(fp.name)
+    except Exception as e:
+        if type(e) is NotImplementedError:
+            raise e
+        else:
+            raise(exceptions.EmptyTableError("Cannot render an empty mat file"))
+
+    if not workbook:
+        raise(exceptions.EmptyTableError("Cannot render an empty mat file"))
```

Comment:
Just checked the `scipy.io.loadmat()` [doc](https://docs.scipy.org/doc/scipy-0.19.1/reference/generated/scipy.io.loadmat.html), it always returns an dictionary, can be `empty` but never `None`. So instead of "Cannot render an empty mat file", can we use something like "Warning: the .mat file is empty"?